lib/rspec/mocks/proxy.rb

```diff
@@ -33,20 +31,6 @@ def initialize(object, order_group, options={})
         @method_doubles = Hash.new { |h, k| h[k] = MethodDouble.new(@object, k, self) }
       end
 
-      # @private
-      def ensure_can_be_proxied!(object)
-        return unless object.is_a?(Symbol) || object.frozen?
-        return if object.nil?
-
-        msg = "Cannot proxy frozen objects"
-        if Symbol === object
-          msg << ". Symbols such as #{object} cannot be mocked or stubbed."
-        else
-          msg << ", rspec-mocks relies on proxies for method stubbing and expectations."
-        end
-        raise ArgumentError, msg
-      end
-
```

Comment:
Restore this, if you want to relax the `object.frozen?` check do so but Symbols can't be proxied and was the original source of this code.

Reply:
As noted in another comment, I changed this logic to catch the resulting TypeError, but I will revert to this way.

Reply:
This still need to restore the original symbol check.

Reply:
Ok, so just so I'm clear. You want to maintain the check for Symbol in this precise location, even though it's covered by the exception handling in method_double? Are you ok with not checking `#frozen?` for any classes here? I'm confused because in https://github.com/rspec/rspec-mocks/pull/1401#discussion_r834210635 you said I was missing handling for non-Symbol/String frozen objects, which is the entire point of this PR.

Reply:
OK, got it now.